File: trunk/resources/scop_1.5.1/java/examples/XMLServer.java
// XMLServer.java - DMI - 25-12-02

import java.io.*;

public class XMLServer
{
	public static void main(String[] args)
	{
		SCOPXML scop;
		Vertex v, w;

		scop = new SCOPXML("localhost", "XMLServer");
		while(true)
		{
			v = scop.get_request();
			w = Vertex.pack(combi(v.extract_int(0), v.extract_int(1)));
			scop.send_reply(w);
		}
		// scop.close();
	}

	private static int combi(int n, int k)
	{
		int result = 1;
		if(k > n || k < 0)
			return 0;

		for(int i = 0; i < k; i++)
			result *= n - i;

		for(int i = 1; i <= k; i++)
			result /= i;

		return result;
	}
}

File: trunk/resources/scop_1.5.1/python/examples/xml_server.py
#!/usr/bin/python
# xml_server.cpp - DMI - 11-11-03

import scop

def combi(n, k):
	result = 1;
	if k > n or k < 0:
		return 0	
	for i in range(k):
		result *= n - i
	for i in range (1, k + 1):
		result /= i
	return result

sock = scop.scop_open("localhost", "xml_server")
while 1:
	v = scop.scop_get_request(sock)
	w = combi(v[0], v[1])
	scop.scop_send_reply(sock, w)
scop.scop_close(sock)
